lib/spack/spack/cmd/checksum.py

```diff
@@ -57,7 +65,8 @@ def checksum(parser, args):
             tty.die("Could not find any versions for {0}".format(pkg.name))
 
     version_lines = spack.stage.get_checksums_for_versions(
-        url_dict, pkg.name, keep_stage=args.keep_stage)
+        url_dict, pkg.name, keep_stage=args.keep_stage,
+        batch=(args.yes or len(url_dict) > 1))
```

Comment:
I think this should be:
```suggestion
        batch=(args.yes or len(args.versions) > 1))
```
Otherwise we will automatically download all versions whenever we find more than 1. This could lead to hundreds of downloads.